File: tests/omnia/data/cache/CachedLongTest.java
package omnia.data.cache;

import static com.google.common.truth.Truth.assertThat;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class CachedLongTest {

  @Test
  public void cache_didReturnSimpleLongCache() {
    assertThat(CachedLong.cache(() -> 132L)).isInstanceOf(SimpleLongCacher.class);
  }
}

File: tests/omnia/data/cache/CachedLongTest.kt
package omnia.data.cache

import com.google.common.truth.Truth.assertThat
import kotlin.test.Test

class CachedLongTest {

  @Test
  fun cache_didReturnSimpleLongCache() {
    assertThat(CachedLong.cache { 132L }).isInstanceOf(SimpleLongCacher::class.java)
  }
}